File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string code) {
    string ret;
    int mode = 0;
    for(int i = 0; i < code.size(); i++) {
        if(mode == 0) {
            if(code[i] != '1' && i % 2 == 0) ret += code[i];
            if(code[i] == '1') mode = 1;
        } else {
            if(code[i] != '1' && i % 2 != 0) ret += code[i];
            if(code[i] == '1') mode = 0;
        }
    }
    
    if(ret.empty()) return "EMPTY";
    return ret;
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        StringBuilder answer = new StringBuilder();
        int mode = 0;

        for (int i = 0; i < code.length(); i++) {
            if (mode == 0) {
                if (code.charAt(i) != '1' && i % 2 == 0) answer.append(code.charAt(i));
                if (code.charAt(i) == '1') mode = 1;
            } else {
                if (code.charAt(i) != '1' && i % 2 != 0) answer.append(code.charAt(i));
                if (code.charAt(i) == '1') mode = 0;
            }
        }

        return answer.length() == 0 ? "EMPTY" : answer.toString();
    }
}